autorelease/trigger.py

```diff
@@ -0,0 +1,127 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This module handles automatically running releasetool tag against all pending PRs."""
+
+import importlib
+
+from autorelease import common, github, kokoro, reporter
+
+LANGUAGE_ALLOWLIST = []
+ORGANIZATIONS_TO_SCAN = ["googleapis", "GoogleCloudPlatform"]
+
+
+def process_issue(kokoro_session, gh: github.GitHub, issue: dict, result) -> None:
+    # Reify the "issue" into a full pull request object from github. This
+    # is necessary because github gives us back an issue object, but it
+    # doesn't contain all of the PR info.
+    pull = gh.get_url(issue["pull_request"]["url"])
+
+    # Before doing any processing, check to make sure the PR was actually merged.
+    # "closed" PRs can be merged or just closed without merging.
+    if not pull.get("merged_at"):
+        result.skipped = True
+        result.print("Closed, not merged, skipping.")
+        # Remove the label so we don't continue processing it.
+        gh.update_pull_labels(
+            pull, add=["autorelease: closed"], remove=["autorelease: pending"]
+        )
+        return
+
+    # Determine language.
+    lang = common.guess_language(gh, pull["base"]["repo"]["full_name"])
+
+    # As part of the migration to release-please tagging, cross-reference the
+    # language against an allowlist to allow migrating language-by-language.
+    if lang not in LANGUAGE_ALLOWLIST:
+        result.skipped = True
+        result.print(f"Language {lang} not in allowlist, skipping.")
+        return
+
+    language_module = importlib.import_module(f"releasetool.commands.tag.{lang}")
+    package_name = language_module.package_name(pull)
```

Comment:
If you want to avoid defining `package_name()` for the 6 other languages:
```py
get_package_name = getattr(language_module, 'package_name', lambda x: None)
package_name = get_package_name(pull)
```

Reply:
I kinda like the consistency of having the module define `package_name` for all the languages, but could possibly be swayed otherwise. I do like that I can have an explicit test on each language module for its response versus potentially testing a `get_package_name(pull, lang)` method in the `autorelease.trigger` module that is actually testing the individual language modules.

Reply:
SG.